R/connection_spark.R

```diff
@@ -217,6 +238,13 @@ spark_connect <- function(master,
   # register mapping tables for spark.ml
 
   register_mapping_tables()
+  
+  if (method == "qubole") {
```

Comment:
Instead of special-casing here for `qubole`, create a new dispatcher function using S3 R methods. This is accomplished as follows:

1) In the bottom of this file add the main S3 dispatcher:

```r
initialize_method <- function(method, scon) {
   UseMethod("initialize_method")
}
```

2) Also add the default behavior in this file which to not do anything:

```r
initialize_method.default <- function(method, scon) {
   scon
}
```

3) Create a method object and dispatch the function by replacing the `if (method == "qubole")` with:

```r
scon <- initialize_method(structure(method, class = "qubole"), scon)
```

4) Implement your own initializer, in `qubole_connection.R` add this function:

```r
initialize_method.qubole <- function(method, scon) {
  scon$config$sparklyr.web.spark <- invoke(scon$state$spark_context,
                                             "getSparkUIURL")
  scon$config$sparklyr.web.yarn <- system("/usr/lib/zeppelin/bin/get_yarn_url.sh",
                                            intern = TRUE)
  scon
}
```